cmd/snap/inhibit.go

```diff
@@ -82,6 +75,12 @@ func waitWhileInhibited(ctx context.Context, cli *client.Client, snapName string
 		return err
 	}
 	inhibited := func(ctx context.Context, hint runinhibit.Hint, inhibitInfo *runinhibit.InhibitInfo) (cont bool, err error) {
+		if hint == runinhibit.HintInhibitedForRemove {
+			return false, errInhibitedForRemove
+		}
+		if !features.RefreshAppAwareness.IsEnabled() {
+			return true, errOngoingSnapRefresh
```

Comment:
I don't understand this change, before we would simply let the snap run even if there was a lock, no?

Reply:
Previously, `snap run` would almost always fail to run due to missing current symlink, because [`backend.UnlinkSnap`](https://github.com/snapcore/snapd/blob/fc53358689e6ef8d32528c9b11803e4340046531/overlord/snapstate/backend/link.go#L341) (called in `unlink-current-snap`) inhibits the snap then removes the current symlink (except of course for the tiny window between inhibition and the removal of the current symlink).

I changed the logic slightly to make it predictable, now it would always fail to start if the snap is inhibited.